docs/BUILDING.md

```diff
@@ -88,3 +88,32 @@ To enable testing, run `make check`.
 
 **Note:** If make check runs 0 tests, you likely need the configure options `--enable-unit` and `--enable-integration`. See [Configure Options](#configure-options)
 for more details.
+
+## Running tests in a container
+
+Sometimes it is useful to be able to run tests in a fresh environment where everything is configured by default.
+Using a container is a light way to ensure the environment of the test is quite clean.
+There are several containers provided in [tpm2-software-container](https://github.com/tpm2-software/tpm2-software-container) project.
+
+For example here is a command which builds tpm2-pkcs11 and runs continuous integration tests in a Fedora 32 container with [podman](https://podman.io/):
+
+```sh
+podman run --rm -v "$(pwd):/workspace/tpm2-pkcs11" --env-file .ci/docker.env \
```

Comment:
Why podman and not docker, or provide both? Are they just "drop in" replacements like clang and gcc? I've never bumped into podman, so I'd like to document to the tooling with the larger user base.

Reply:
I am probably biased by the tweet from https://twitter.com/rhatdan (a Red Hat employee who contributed a lot to SELinux and Docker before working on OCI Containers and Podman, Buildah, etc.).

Personally I do not like the concept of "being a member of the `docker` group is exactly the same as being `root`", which is a major drawback of Docker. Instead of using virtual machines to contain Docker, I prefer using a solution which leverages user namespaces so that running containers does not enable running things as root. Podman exists since 2018 so it is quite recent, but with Red Hat pushing for it (with some systemd integration for example) and with the fact that its command-line interface is compatible with Docker's one, I feel like it can be documented as an alternative to Docker.

Anyway, I can add a command-line with `sudo docker` too, to emphasize that being able to launch a container with `docker` is really like being able to run any command as `root`. Would you agree with this?

Reply:
> You don't need sudo docker, just docker

No. If `docker` can be used, it can be a security issue, as documented in https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user :

    The docker group grants privileges equivalent to the root user.
    For details on how this impacts security in your system, see Docker Daemon Attack Surface.

I do not want to advocate a weak security configuration without a clear warning that this is a weak security configuration and advice about stronger alternatives.

Reply:
I added a message about this:

```sh
sudo docker run --rm -v "$(pwd):/workspace/tpm2-pkcs11" --env-file .ci/docker.env \
    --workdir /workspace/tpm2-pkcs11 -ti ghcr.io/tpm2-software/fedora-32 \
    bash -c .ci/docker.run
# NB. sudo can be omitted if the user is added to the "docker" group, but this
# configuration impacts security in your system, as documented in
# https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user
```

Would such a message suits you?

Reply:
Thats fine